dashboard/app/models/stage.rb

```diff
@@ -37,8 +37,16 @@ def unplugged?
   end
 
   def localized_title
+    # The standard case for localized_title is something like "Stage 1: Maze".
+    # In the case of lockable stages, we don't want to include the Stage 1
+    return I18n.t("data.script.name.#{script.name}.#{name}") if lockable
+
     if script.stages.to_a.many?
-      I18n.t('stage_number', number: position) + ': ' + I18n.t("data.script.name.#{script.name}.#{name}")
+      # Because lockable stages aren't numbered, our stage number is actually our
+      # position, minus the number of lockable stages preceeding us
+      stage_number = position - script.stages.to_a[0, position].count(&:lockable)
```

Comment:
luckily, n is also always quite small (<100)

I'm not sure we can actually cache easily here, as we need more than just total_lockable - we need how many are lockable before us. If stages 2 and 4 are lockable, this means the number is 1 for stage 3, but 2 for stage 5.